virtcontainers/pkg/annotations/annotations.go

```diff
@@ -51,36 +66,135 @@ const (
+	//
+
 	// DefaultMemory is a sandbox annotation for the memory assigned for a VM by the hypervisor.
 	DefaultMemory = kataAnnotHypervisorPrefix + "default_memory"
 
-	// KernelParams is a sandbox annotation for passing additional guest kernel parameters.
-	KernelParams = kataAnnotHypervisorPrefix + "kernel_params"
+	// MemSlots is a sandbox annotation to specify the memory slots assigned to the VM by the hypervisor.
+	MemSlots = kataAnnotHypervisorPrefix + "memory_slots"
+
+	// MemOffset is a sandbox annotation that specifies the memory space used for nvdimm device by the hypervisor.
+	MemOffset = kataAnnotHypervisorPrefix + "memory_offset"
+
+	// MemPrealloc is a sandbox annotation that specifies the memory space used for nvdimm device by the hypervisor.
+	MemPrealloc = kataAnnotHypervisorPrefix + "enable_mem_prealloc"
+
+	// EnableSwap is a sandbox annotation to enable swap of vm memory.
+	// The behaviour is undefined if mem_prealloc is also set to true
+	EnableSwap = kataAnnotHypervisorPrefix + "enable_swap"
+
+	// HugePages is a sandbox annotation to specify if the memory should be pre-allocated from huge pages
+	HugePages = kataAnnotHypervisorPrefix + "enable_hugepages"
+
+	// FileBackedMemRootDir is a sandbox annotation to soecify file based memory backend root directory
+	FileBackedMemRootDir = kataAnnotHypervisorPrefix + "file_mem_backend"
+
+	//
+	//	Shared File System related annotations
+	//
+
+	// Msize9p is a sandbox annotation to specify as the msize for 9p shares
+	Msize9p = kataAnnotHypervisorPrefix + "msize_9p"
+
+	// SharedFs is a sandbox annotation to specify the shared file system type, either virtio-9p or virtio-fs.
+	SharedFS = kataAnnotHypervisorPrefix + "shared_fs"
+
+	// VirtioFSDaemon is a sandbox annotations to specify virtio-fs vhost-user daemon path
+	VirtioFSDaemon = kataAnnotHypervisorPrefix + "virtio_fs_daemon"
+
+	// VirtioFSCache is a sandbox annotation to specify the cache mode for fs version cache or "none"
+	VirtioFSCache = kataAnnotHypervisorPrefix + "virtio_fs_cache"
+
+	// VirtioFSCacheSize is a sandbox annotation to specify the DAX cache size in MiB
+	VirtioFSCacheSize = kataAnnotHypervisorPrefix + "virtio_fs_cache_size"
+
+	// VirtioFSExtraArgs is a sandbox annotation to pass options to virtiofsd daemon
+	VirtioFSExtraArgs = kataAnnotHypervisorPrefix + "virtio_fs_extra_args"
+
+	//
+	//	Block Device related annotations
+	//
+
+	// BlockDeviceDriver specifies the driver to be used for block device either VirtioSCSI or VirtioBlock
+	BlockDeviceDriver = kataAnnotHypervisorPrefix + "block_device_driver"
+
+	// DisableBlockDeviceUse  is a sandbox annotation that disallows a block device from being used.
+	DisableBlockDeviceUse = kataAnnotHypervisorPrefix + "disable_block_device_use"
+
+	// EnableIOThreads is a sandbox annotation to enable IO to be processed in a separate thread.
+	// Supported currently for virtio-scsi driver.
+	EnableIOThreads = kataAnnotHypervisorPrefix + "enable_iothreads"
```

Comment:
It would be nice that golang could use a macro so enable_iothreads cold be reused  so we dont have to modify this in case a config name changes :(